sct-commons/src/test/java/org/lfenergy/compas/sct/commons/util/UtilsTest.java

```diff
@@ -436,4 +438,107 @@ void lnClassEquals_when_lnClass_has_more_than_one_value_should_throw_exception()
         assertThatThrownBy(() -> Utils.lnClassEquals(lnClass1, lnClass2))
             .isInstanceOf(IllegalArgumentException.class);
     }
+
+    @Test
+    void sequence_should_return_range_of_numbers(){
+        // Given
+        long startInclusive = 1;
+        long endInclusive = 3;
+        // When
+        PrimitiveIterator.OfLong result = Utils.sequence(startInclusive, endInclusive);
+        // Then
+        assertThat(result.next()).isEqualTo(1);
+        assertThat(result.next()).isEqualTo(2);
+        assertThat(result.next()).isEqualTo(3);
+        assertThat(result.hasNext()).isFalse();
+        assertThatThrownBy(result::nextLong).isInstanceOf(NoSuchElementException.class);
+    }
+
+    @Test
+    void sequence_should_return_empty_iterator(){
+        // Given
+        long startInclusive = 1;
+        long endInclusive = 0;
+        // When
+        PrimitiveIterator.OfLong result = Utils.sequence(startInclusive, endInclusive);
+        // Then
+        assertThat(result.hasNext()).isFalse();
+        assertThatThrownBy(result::nextLong).isInstanceOf(NoSuchElementException.class);
+    }
+
+    @Test
+    void sequence_should_throw_exception_when_MAX_VALUE(){
+        // Given
+        long startInclusive = 1;
+        long endInclusive = Long.MAX_VALUE;
+        // When & Then
+        assertThatThrownBy(() -> Utils.sequence(startInclusive, endInclusive))
+            .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void macAddressSequence_should_return_range_of_macAddresses(){
+        // Given
+        String startInclusive = "01-0C-CD-01-00-FE";
+        String endInclusive = "01-0C-CD-01-01-01";
+        // When
+        Iterator<String> result = Utils.macAddressSequence(startInclusive, endInclusive);
+        // Then
+        assertThat(result.next()).isEqualTo("01-0C-CD-01-00-FE");
+        assertThat(result.next()).isEqualTo("01-0C-CD-01-00-FF");
+        assertThat(result.next()).isEqualTo("01-0C-CD-01-01-00");
+        assertThat(result.next()).isEqualTo("01-0C-CD-01-01-01");
+        assertThat(result.hasNext()).isFalse();
+        assertThatThrownBy(result::next).isInstanceOf(NoSuchElementException.class);
+    }
+
+    @ParameterizedTest
+    @CsvSource({"123456789ABC,12-34-56-78-9A-BC", "AA,00-00-00-00-00-AA"})
+    void longToMacAddress_should_convert_long_to_mac_address(String hex, String expected){
```

Comment:
Could give directly a long instead of given hex; which is converted in to long (in //Given) before process

Reply:
OK.
I changed to this
```java
    @CsvSource({"0x123456789ABC,12-34-56-78-9A-BC", "0xAA,00-00-00-00-00-AA"})
    void longToMacAddress_should_convert_long_to_mac_address(long macAddress, String expected){
...
}
```